forms/core/Application.php

```diff
@@ -644,37 +595,40 @@ public static function generateConnectPage()
             $clients = Application::$CampaignMonitor->get_clients( $auth );
             Settings::add( 'campaign_monitor_clients', $clients );
 
-	        if ( ! empty( $clients ) ) {
-		        if ( \is_array( $clients ) ) {
-			        if (count( $clients ) === 1 && !empty($clients[0])) {
+	        if (!empty($clients)) {
+		        if (is_array($clients)) {
+			        if (count($clients) === 1 && !empty($clients[0])) {
 				        $CID = $clients[0]->ClientID;
 				        Settings::add( 'default_client', $CID );
 			        }
 		        } else {
-
-			        if ( ! empty( $clients->Message ) && strpos($clients->Message, 'Authorization') !== false ) {
-				        $message = 'To refresh your credentials please click on the "Disconnect Account" button';
-				        $message .= ' and then follow the on screen instruction to re-connect again!';
-				        $message = \urlencode( $message );
-			            ?>
-                        <div class="update-nag error is-dismissible notice">
-                            <p>
-                                There seems to be a problem with your credentials please disconnect and reconnect your account on the
-                                <a href="<?php echo  get_admin_url() . '/admin.php?page=campaign_monitor_settings_page&notice[description]='.filter_var($message,FILTER_SANITIZE_STRING).'&notice[title]=Notice!' ?>"> Campaign Monitor Settings </a> page
-                            </p>
-                        </div>
-                        <?php
+			        if (isset($clients->error)) {

```

Comment:
Could you make sure that this check is working as intended? Would we have error other than authorization issue?

Reply:
The other error (from API) have the object structure { code: '', message: '' } so it shouldn't hit that.